File: docs/guides/go/src/useConditionlessRule.go
package main

import (
	"context"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
	"github.com/algolia/algoliasearch-client-go/v4/algolia/utils"
)

func useConditionlessRule() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	objectID := "a-rule-id"

	rule := search.NewRule(
		objectID,
		*search.NewConsequence( /* Set relevant consequence */ ),
	). // Set validity (optional)
		SetValidity(
			[]search.TimeRange{{
				From:  utils.ToPtr(int64(1_688_774_400)),
				Until: utils.ToPtr(int64(1_738_972_800)),
			}},
		)

	_, err = client.SaveRule(context.Background(), "<YOUR_INDEX_NAME>", objectID, rule, nil)
	if err != nil {
		panic(err)
	}
}

File: docs/guides/python/useConditionlessRule.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.consequence import Consequence
from algoliasearch.search.models.rule import Rule
from algoliasearch.search.models.time_range import TimeRange

_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


object_id = "a-rule-id"

rule = Rule(
    object_id=object_id,
    consequence=Consequence(
        # Set relevant consequence
    ),
)

# Set validity (optional)
rule.validity = [
    TimeRange(
        var_from=1688774400,
        until=1738972800,
    ),
]

_client.save_rule(
    index_name="<YOUR_INDEX_NAME>",
    object_id=object_id,
    rule=rule,
)
